Codeinwp/otter-blocks
# Allow saving in FSE for Form Options

Closes #1802


### Summary


#### [UPDATED]

A straightforward solution to this problem is to insert a dummy block and then delete it immediately. This will mark the block as dirty, thus enabling the Post/Page saving. The functionality is wrapped in a `debounce` function, which prevents multiple events from triggering simultaneously. 

ℹ️ This works reliably in Post/Pages/FSE/Widgets/Customizer 🔥 

#### [OLD SOLUTION]
Usually, the Save button is enabled when you make a change to the content of the post/page. The problem arises when some functionality is not tied to the content but is triggered by the `Save` button.

Form Options live in the WP Options. When a user changes something in the Form the Mail, the values must be saved in WP Options to take effect. They are not tied to the block attributes, thus making them invisible for the Editor to detect changes.

After some heavy investigation regarding the saving mechanism in the Gutenberg editor, a simple trick to enable the saving is to make a request using the function `dispatch( 'core' ).editEntityRecord`.

In this PR, we implement a function that makes this request by adding an unregistered meta value to the entity. Since it is unregistered, the value is not saved anymore, but making the request marks the post/page as `dirty`, thus enabling the `Save` button.

⚠️ This is not working with Widgets on the widget page.

### Screenshots 



https://github.com/Codeinwp/otter-blocks/assets/17597852/7d0bf192-3b90-4353-b646-37c307949560


----

### Test instructions


1. Use an FSE theme.
2. Add a form inside of the templates/pages.
3. Save the form as it is, then change some values in the Form Options.
4. `Save` button should be available.

⚠️ The `Save` will be available after 3 seconds without modification. As long as you type or change something in the Form options within 3 seconds, it will not trigger.

⚠️ You should notice `Form options have been saved`, which confirms the save request.



---- 

### Checklist before the final review
